src/hotspot/share/oops/symbolHandle.hpp

```diff
@@ -76,6 +90,19 @@ class SymbolHandleBase : public StackObj {
     Symbol::maybe_decrement_refcount(_temp);
   }
 
+  // Keep this symbol alive for some time to allow for reuse.
+  // Temp symbols for the same string can often be created in quick succession,
+  // and this queue allows them to be reused instead of churning.
+  void add_to_cleanup_delay_queue(Symbol* sym) {
+    sym->increment_refcount();
+    STATIC_ASSERT(is_power_of_2(CLEANUP_DELAY_MAX_ENTRIES)); // allow modulo shortcut
+    uint i = Atomic::add(&_cleanup_delay_index, 1u) & (CLEANUP_DELAY_MAX_ENTRIES - 1);
```

Comment:
I disagree with @shipilev .  Just use `% CLEANUP_DELAY_MAX_ENTRIES`.  This is an entirely
unreasonable amount of code to explicitly provide a micro-optimization that any non-stupid compiler
will do for us anyway. At most, add a comment to the definition of that constant that being a power
of 2 benefits the ring-buffer.  But I wouldn't even bother with that.